rest-service/manager_rest/test/test_import_resolver.py

```diff
@@ -58,195 +42,52 @@ def _update_provider_context(self, resolver_section=None):
-            ]
-        }
-        resolver_impl = _get_instance_class_path(DefaultImportResolver())
-        resolver_section = self._create_resolver_section(
-            resolver_impl=resolver_impl,
-            resolver_params=params)
-        self._test_resolver(resolver_section=resolver_section,
-                            expected_resolver_impl=resolver_impl,
-                            expected_params=params)
-
-    def test_custom_resolver(self):
-        # import_resolver section is specified in provider context
-        # points to some custom resolver
-        params = {
-            'param1': 'value1'
-        }
-        resolver_impl = _get_instance_class_path(CustomResolver())
-        resolver_section = self._create_resolver_section(
-            resolver_impl=resolver_impl,
-            resolver_params=params)
-        self._test_resolver(resolver_section=resolver_section,
-                            expected_resolver_impl=resolver_impl,
-                            expected_params=params)
-
-    def test_resolver_already_in_app(self):
-        params = {
-            'param1': 'value1'
-        }
-        resolver_impl = _get_instance_class_path(CustomResolver())
-
-        resolver_section = self._create_resolver_section(
-            resolver_impl=resolver_impl,
-            resolver_params=params)
-
-        # update provider context with resolver and
-        self._test_resolver(resolver_section=resolver_section,
-                            expected_resolver_impl=resolver_impl,
-                            expected_params=params)
+    def test_upload_blueprint_with_resolver(self, mock_parse_dsl):
+
+        resolver_section = 'mock resolver section'
+        create_import_resolver_inputs = []
 
+        def mock_create_import_resolver(resolver_section):
+            create_import_resolver_inputs.append(resolver_section)
+            return 'mock expected import resolver'
+
+        mock_parse_dsl.return_value = None
+        with mock.patch(
+                'dsl_parser.utils.create_import_resolver',
+                new=mock_create_import_resolver):
+            self._update_provider_context(resolver_section)
+            self.put_file(*self.put_blueprint_args())
+
+        # asserts
+        mock_parse_dsl.assert_called_once_with(
+            mock.ANY, mock.ANY, 'mock expected import resolver')
+        self.assertEqual(create_import_resolver_inputs[0],
+                         resolver_section)
+        self.assertEqual(self.app.application.resolver,
+                         'mock expected import resolver')
+
+    def test_resolver_update_in_app(self):
+        # the next test updates the resolver in the app
+        self.test_upload_blueprint_with_resolver()
+        curr_app_resolver = self.app.application.resolver
         # upload blueprint again and check that
         # the expected resolver passed to the parser
         with mock.patch('dsl_parser.tasks.parse_dsl') as mock_parse_dsl:
```

Comment:
I think you already test the same thing in "test_upload_blueprint_with_resolver"